src/main/java/org/elasticsearch/script/ScriptModes.java

```diff
@@ -0,0 +1,215 @@
+    private static void processEngineSpecificSettings(Settings settings, Map<String, ScriptEngineService> scriptEngines, List<String> processedSettings, Map<String, ScriptMode> scriptModes) {
+        Map<String, Settings> langGroupedSettings = settings.getGroups(ENGINE_SETTINGS_PREFIX, true);
+        for (Map.Entry<String, Settings> langSettings : langGroupedSettings.entrySet()) {
+            //read engine specific settings that refer to a non existing script lang will be ignored
+            ScriptEngineService scriptEngineService = scriptEngines.get(langSettings.getKey());
+            if (scriptEngineService != null) {
+                for (ScriptType scriptType : ScriptType.values()) {
+                    for (ScriptedOp scriptedOp : ScriptedOp.values()) {
+                        String scriptTypePrefix = scriptType + ".";
+                        ScriptMode scriptMode = getScriptedOpMode(langSettings.getValue(), scriptTypePrefix, scriptedOp);
+                        if (scriptMode != null) {
+                            processedSettings.add(scriptTypePrefix + scriptedOp + ": " + scriptMode);
+                            addScriptMode(scriptEngineService, scriptType, scriptedOp, scriptMode, scriptModes);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private static void processDisableDynamicDeprecatedSetting(Settings settings, Map<String, ScriptEngineService> scriptEngines,
+                                                               List<String> processedSettings, Map<String, ScriptMode> scriptModes, ESLogger logger) {
+        //read deprecated disable_dynamic setting, apply only if none of the new settings is used
+        String disableDynamicSetting = settings.get(ScriptService.DISABLE_DYNAMIC_SCRIPTING_SETTING);
+        if (disableDynamicSetting != null) {
+            if (processedSettings.isEmpty()) {
+                ScriptService.DynamicScriptDisabling dynamicScriptDisabling = ScriptService.DynamicScriptDisabling.parse(disableDynamicSetting);
+                switch(dynamicScriptDisabling) {
+                    case EVERYTHING_ALLOWED:
+                        addGlobalScriptTypeModes(scriptEngines.keySet(), ScriptType.INDEXED, ScriptMode.ENABLE, scriptModes);
+                        addGlobalScriptTypeModes(scriptEngines.keySet(), ScriptType.INLINE, ScriptMode.ENABLE, scriptModes);
+                        break;
+                    case ONLY_DISK_ALLOWED:
+                        addGlobalScriptTypeModes(scriptEngines.keySet(), ScriptType.INDEXED, ScriptMode.DISABLE, scriptModes);
+                        addGlobalScriptTypeModes(scriptEngines.keySet(), ScriptType.INLINE, ScriptMode.DISABLE, scriptModes);
+                        break;
+                }
+            } else {
+                logger.warn("ignoring [{}] setting as conflicting scripting settings have been specified: {}", ScriptService.DISABLE_DYNAMIC_SCRIPTING_SETTING, processedSettings);
```

Comment:
that would be an easy way to get rid of the logger too... :) I discussed this with @clintongormley and we agreed on only logging and ignoring old settings, but sure we can throw an exception and it will make the node startup fail.